### PR TITLE
V2 plugin startup

### PR DESCRIPTION
**Pull request by josiahhaswell**

### CODE REVIEW

File: kernel-core/src/main/java/io/sunshower/kernel/core/actions/WriteKernelModuleListPhase.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import java.io.OutputStreamWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
```
Comment (lisagumerman) [inline kernel-core/src/main/java/io/sunshower/kernel/core/actions/WriteKernelModuleListPhase.java:23]: isn't this the wrong logger?

File: kernel-core/src/main/java/io/sunshower/kernel/core/actions/plugin/PluginStartTask.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   * be looking at the state while a plugin is starting--we need to listen for the relevant module
+   * lifecycle events dispatched by the PluginContext
+   */
+  @Override
+  @SuppressWarnings("PMD.AvoidBranchingStatementAsLastInLoop")
+  public TaskValue run(Scope scope) {
+    synchronized (this) {
+      val module = manager.getModule(coordinate);
+      val currentState = module.getLifecycle().getState();
+      if (!currentState.isAtLeast(Lifecycle.State.Active)) { // // TODO: 11/11/19 handle failed
```
Comment (lisagumerman) [inline kernel-core/src/main/java/io/sunshower/kernel/core/actions/plugin/PluginStartTask.java:36]: What's the failure going to look like? A whole rollback a la Troposphere or just a log line?

File: kernel-gyre/gyre-api/src/main/java/io/sunshower/gyre/ReverseSubgraphTransformation.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Override
+  public Graph<E, V> apply(Graph<E, V> graph, Predicate<E> edgeFilter, Predicate<V> nodeFilter) {
+    val newgraph = graph.createNew();
+    val stack = new Stack<Pair<E, V>>();
+    stack.push(Pair.of(null, root));
+    newgraph.add(root);
+    while (!stack.isEmpty()) {
+      val current = stack.pop();
+      val neighbors = graph.getDependents(current.snd, edgeFilter);
```
Comment (lisagumerman) [inline kernel-gyre/gyre-api/src/main/java/io/sunshower/gyre/ReverseSubgraphTransformation.java:27]: Thanks for renaming this, btw

**Review by lisagumerman — APPROVED** [review]

File: kernel-core/src/main/java/io/sunshower/kernel/core/actions/plugin/PluginStartTask.java (hunk 2 above)
Comment (josiahhaswell) [reply kernel-core/src/main/java/io/sunshower/kernel/core/actions/plugin/PluginStartTask.java:36]: This is for if a previous module has failed.  I'm not sure what we should or can do in that situation.

**Pull request merged**